### PR TITLE
Make crash notes and recommendations compatible with the new data model

### PR DESCRIPTION
**Pull request by roseeichelmann**
## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/18045

## Testing
**URL to test:** 
1. Check out this branch and follow instructions in https://github.com/cityofaustin/atd-vz-data/pull/1440 to start the data model, apply migrations and run the CRIS import ETL
2. You will need to open up your hasura console or whatever database tool you prefer. Go to the `recommendations` table, find a recommendation. Copy the `atd_txdot_crashes_crash_id`. In the `crashes` table use that to find the `id` pkey for that crash, copy that. Now in the recommendations table edit the record and put the `id` pkey for the crash in the `crash_id` column of the recommendation.
3. Now in the VZE go to that crash, you should see the recommendation pop up for it now. Test editing diff parts of the recommendation.
4. Now create some new notes for the crash and edit/delete them.
5. You can also do the same thing you did in steps 2-3 but for a note.

Sorry these test instructions are weird, just wanted to get this PR aligned as much as possible with the new data model so that next we will just backfill the notes/recommendations with the correct `id` pkey in the `crash_id` column and then will be able to delete the `atd_txdot_crashes_crash_id` we have here all together.

**Steps to test:**

---
#### Ship list
- [ ] Check migrations for any conflicts with latest migrations in master branch
- [ ] Confirm Hasura role permissions for necessary access
- [ ] Code reviewed
- [ ] Product manager approved

### CODE REVIEW

File: atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+alter table recommendations rename column crash_id to atd_txdot_crashes_crash_id;
+
+alter table recommendations
+    add column crash_id integer,
+    add constraint recommendations_crashes_id_fkey foreign key (
+        crash_id
+    ) references crashes (id),
+    drop constraint recommendations_crash_id_key,
+    add constraint recommendations_crash_id_key unique (crash_id),
+    alter column atd_txdot_crashes_crash_id drop not null;
```
Comment (roseeichelmann) [inline atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql:10]: @johnclary do i need to add the updates i made to this migration to the `sql_templates` files?

File: atd-vze/src/Components/Notes/Notes.js [hunk 2]
```diff
@@ -35,19 +36,11 @@ const Notes = ({
   // get current users email
   const userEmail = localStorage.getItem("hasura_user_email");
 
-  // fetch data from database using graphQL query
-  const { loading, error, data, refetch } = useQuery(GET_NOTES, {
```
Comment (roseeichelmann) [inline atd-vze/src/Components/Notes/Notes.js:39]: now we get the notes from the main crash query :pray:

File: atd-vze/src/Components/Notes/Notes.js [hunk 3]
```diff
@@ -56,7 +49,7 @@ const Notes = ({
     addNote({
       variables: {
         note: newNote,
-        recordId: recordId,
+        crashPk: crashPk,
```
Comment (roseeichelmann) [inline atd-vze/src/Components/Notes/Notes.js:52]: now adding new notes and querying notes using our internal crash id pkey

File: atd-vze/src/Components/Notes/Notes.js [hunk 4]
```diff
@@ -198,100 +191,101 @@ const Notes = ({
...
-                          format(parseISO(row[field]), "MM/dd/yyyy")
-                        ) : (
-                          row[field]
-                        )}
-                      </td>
-                    );
-                  })}
-                  {/* display edit button if row was created by current user,
+            {notes &&
+              notes.map(row => {
```
Comment (roseeichelmann) [inline atd-vze/src/Components/Notes/Notes.js:195]: this is really the only change for the rest of this file

File: atd-vze/src/queries/recommendations.js [hunk 5]
```diff
@@ -1,27 +1,7 @@
...
-        id
-        partner_id
-        recommendation_id
-        atd__coordination_partners_lkp {
-          id
-          coord_partner_desc
-        }
-      }
-    }
+export const GET_RECOMMENDATION_LOOKUPS = gql`
```
Comment (roseeichelmann) [inline atd-vze/src/queries/recommendations.js:3]: only using this for querying the rec lookups now

**Comment by roseeichelmann** [comment]
Also -- i know how confusing and contradicting all the various `id` and `crash_id` and `cris_crash_id` and `atd_txdot_crashes_crash_id`'s are. I think we are gonna talk about ways to name these columns so its less confusing in our next data model meeting. I floated to john the following idea - in the `crashes` table, we have our internal `id` pkey and a column called `cris_crash_id` for the classic crash id we know and love. in other tables we have a column called `crash_pk` that references the `id` of the crashes table, and then a column `cris_crash_id` that references the `cris_crash_id` of crashes if necessary. just a thought :shrug:

File: atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql
Comment (mddilley) [inline atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql:1]: tested these constraints for good measure, and i was not able to save an `id` that doesn't exist in the `crashes_cris` table 👍

**Review by mddilley — APPROVED** [review]
I tested with crash ID `19373803` for recommendations and `19352534` for notes, and I was able to do all of the test steps. Thanks for sharing the plan for backfilling and cleaning up the unneeded columns in these two tables. Sounds great! 🚢 😎

**Comment by mddilley** [comment]
Also, about the `crash_pk` columns, i think that you are on to something. Seeing that used in the code made it clearer to me that it was using the internal (non-CRIS) id. 💯

File: atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql (hunk 1 above)
Comment (johnclary) [reply atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql:10]: yes, please!

File: atd-vze/src/views/Crashes/Recommendations/Recommendations.js [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { useQuery, useMutation } from "@apollo/react-hooks";
 import { Card, CardHeader, CardBody, CardFooter } from "reactstrap";
```
Comment (johnclary) [inline atd-vze/src/views/Crashes/Recommendations/Recommendations.js:3]: I am for the first time noticing these empty `<CardFooter/>` components in the notes and recs, and also all the data table cards 👀 

I'm going to make a snackoo to clean these up bc they are clutter imo 🧹

**Review by johnclary — CHANGES_REQUESTED** [review]
The notes and recommendations are working great! All that's left to do is update that migration template SQL with your edits.

Yes, let's continue the discussion about `crashes.id` vs `crashes.crash_id` and how to best handle this throughout the codebase 🙏

File: atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql (hunk 1 above)
Comment (roseeichelmann) [reply atd-vzd/migrations/default/1715960023005_update_recommendations_and_notes_fks/up.sql:10]: cool just updated that!

**Review by johnclary — APPROVED** [review]
merge merge merge 🙌

**Pull request merged**